torchvision/prototype/datasets/utils/_resource.py

```diff
@@ -141,8 +143,40 @@ def __init__(
         super().__init__(file_name=file_name or pathlib.Path(urlparse(url).path).name, **kwargs)
         self.url = url
         self.mirrors = mirrors
+        self._resolved = False
+
+    @property
+    def resolved(self) -> bool:
+        return self._resolved
+
+    def resolve(self) -> Optional[OnlineResource]:
+        redirect_url = _get_redirect_url(self.url)
+        if redirect_url == self.url:
+            self._resolved = True
+            return self
+
+        meta = {
+            attr.lstrip("_"): getattr(self, attr)
+            for attr in (
+                "file_name",
+                "sha256",
+                "_preprocess",
+                "_loader",
+            )
+        }
+
+        gdrive_id = _get_google_drive_file_id(redirect_url)
+        if gdrive_id:
+            return GDriveResource(gdrive_id, **meta)
+
+        http_resource = HttpResource(redirect_url, **meta)
+        http_resource._resolved = True
+        return http_resource
```

Comment:
Q: Do we consider that this new `http_resource` is itself resolved? (the code doesn't, but should we?)

Reply:
Yes and the code does this. In L173 we set the `_resolved` attribute, which marks this as resolved.

Reply:
oops, I missed that thanks. What about the GDriveResource above?